android-components/components/browser/state/src/test/java/mozilla/components/browser/state/state/TranslationEngineStateTest.kt

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.state
+
+import mozilla.components.concept.engine.translate.DetectedLanguages
+import mozilla.components.concept.engine.translate.Language
+import mozilla.components.concept.engine.translate.TranslationEngineState
+import mozilla.components.concept.engine.translate.TranslationPair
+import mozilla.components.concept.engine.translate.initialFromLanguage
+import mozilla.components.concept.engine.translate.initialToLanguage
+import org.junit.Test
+
+val mockCandidateLanguages = listOf(
+    Language("de", "German"),
+    Language("fr", "French"),
+    Language("en", "English"),
+    Language("es", "Spanish"),
+)
+class TranslationEngineStateTest {
+    @Test
+    fun `initialToLanguage and initialFromLanguage works as expected on the not translated case`() {
+        val detectedLanguages = DetectedLanguages(
+            documentLangTag = "en",
+            supportedDocumentLang = true,
+            userPreferredLangTag = "es",
+        )
+
+        val translationEngineState = TranslationEngineState(
+            detectedLanguages = detectedLanguages,
+            error = null,
+            isEngineReady = true,
+            requestedTranslationPair = null,
+        )
+
+        assert(translationEngineState.initialToLanguage(mockCandidateLanguages)?.code == "es")
```

Comment:
```suggestion
        assert(translationEngineState.initialToLanguage(mockCandidateLanguages)?.code == "es")
```
Perhaps use `assertEquals`

Reply:
Oh this is a good catch. `assert` won't actually fail the tests IIRC, since it isn't a JUnit API. You'll want to use `assertEquals`, `assertTrue`, `assertNull` etc imported from `org.junit`